src/arlo_plugin/provider.py

```diff
@@ -755,6 +805,57 @@ async def discover_devices_impl(self) -> None:
         else:
             self.logger.info(f"Discovered {len(cameras)} cameras")
 
+        """Checks if Security Mode is enabled and if there isn't a primary device already selected. Chooses the first base station or camera and uses that."""
+        if self.mode_enabled == True and (self.mode_device is None or self.mode_device == ""):
```

Comment:
Could be simplified to:
```suggestion
        if self.mode_enabled and not self.mode_device:
```